includes/class-wc-beta-tester-import-export.php

```diff
@@ -0,0 +1,211 @@
+		$this->maybe_import_settings();
+
+		// show error/update messages.
+		if ( ! empty( $this->message ) ) {
+			?>
+			<div class="notice <?php
+				echo ! empty( $this->message['type'] ) ? esc_attr( $this->message['type'] ) : '';
+				?>"><?php echo esc_html( $this->message['message'] ); ?></div>
+			<?php
+		}
+		?>
+		<div class="wrap">
+			<h1><?php echo esc_html( get_admin_page_title() ); ?></h1>
+			<p><?php esc_html_e( 'Export your WooCommerce Settings. The export file should not contain any fields that identify your site or reveal secrets (eg. API keys).', 'woocommerce-beta-tester' ); ?></p>
+			<a href="<?php echo esc_url( $export_url ) ?>" class="button-primary"><?php
+				/* translators Export WooCommerce settings button text. */
+				esc_html_e( 'Export WooCommerce Settings', 'woocommerce-beta-tester' );
+			?></a>
+			<hr />
+			<form method="POST" enctype="multipart/form-data">
+				<?php wp_nonce_field( static::NONCE_ACTION ); ?>
+				<input type="hidden" name="action" value="<?php echo static::IMPORT_ACTION; ?>" />
+				<p><?php esc_html_e( 'Import WooCommerce Settings exported with this tool. Some settings like store address, payment gateways, etc. will need to be configured manually.', 'woocommerce-beta-tester' ); ?></p>
+				<button type="submit" class="button-primary"><?php
+				/* translators Import WooCommerce settings button text. */
+				esc_html_e( 'Import WooCommerce Settings', 'woocommerce-beta-tester' );
+				?></button>
+				<input type="file" name="<?php echo static::IMPORT_FILENAME; ?>" />
+			</form>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Export settings in json format.
+	 */
+	public function export_settings() {
+		if ( ! wp_verify_nonce( $_REQUEST['_wpnonce'], static::NONCE_ACTION ) ) {
+			header( 'HTTP/1.1 403 Forbidden' );
+			exit;
+		}
+
+		$filename = sprintf( 'woocommerce-settings-%s.json', gmdate( 'Ymdgi' ) );
+		wc_set_time_limit( 0 );
+		wc_nocache_headers();
+		header( 'Content-Type: text/csv; charset=utf-8' );
+		header( 'Content-Disposition: attachment; filename=' . $filename );
+		header( 'Pragma: no-cache' );
+		header( 'Expires: 0' );
+		echo wp_json_encode( $this->get_settings() );
+		exit;
+	}
+
+	/**
+	 * Import settings in json format if submitted.
+	 */
+	public function maybe_import_settings() {
+		if ( empty( $_POST ) || empty( $_POST['action'] ) || $_POST['action'] !== static::IMPORT_ACTION ) {
+			return;
+		}
+
+		if ( ! wp_verify_nonce( $_POST['_wpnonce'], static::NONCE_ACTION ) ) {
+			$this->add_message( __( 'Invalid submission', 'woocommerce-beta-tester' ) );
+			return;
+		}
+
+		if ( ! empty( $_FILES[ static::IMPORT_FILENAME ] ) ) {
+			$tmp_file = $_FILES[ static::IMPORT_FILENAME ]['tmp_name'];
+			if ( empty( $tmp_file ) ) {
+				$this->add_message( __( 'No file uploaded.', 'woocommerce-beta-tester' ) );
+				return;
+			}
+
+			if ( is_readable( $tmp_file ) ) {
```

Comment:
This part is nested pretty deeply. Could we pull up a few from the conditionals and transform them into early returning guard clauses (like e.g. the invalid nonce one)? 

Not a blocker.

Reply:
@lsinger I updated the logic here to reduce the nesting. Thanks for the suggestion. It's much easier to connect the message to the condition. Can you take a second pass on this?